### ISSUE TITLE
Gradio app not working inside Next.js

### ISSUE THREAD

**Issue by krrishdholakia** [gradio-app/gradio#2880]
### Describe the bug

Hi, 

I'm embedding my Gradio app inside Next.js.

When I do this, while the component loads - it is not functional. 
<img width="1655" alt="Screen Shot 2022-12-23 at 3 00 47 PM" src="https://user-images.githubusercontent.com/17561003/209310507-10ddd5c0-ee27-4a13-a276-d795df31fe13.png">

This is the error thrown in console log: 

WebSocket connection to 'wss://nextmagictemplateappv2.krrishdholakia.repl.co/queue/join' failed:

I'm not sure why it would be making a queue/join request to my next.js app, when the gradio app is hosted on hf and just being wrapped by the next.js app. 

Any help here would be appreciated. 

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Reproduction

https://nextmagictemplateappv2.krrishdholakia.repl.co/

### Logs

```shell
index.4ac6a7e7.js:34 WebSocket connection to 'wss://nextmagictemplateappv2.krrishdholakia.repl.co/queue/join' failed: 
(anonymous) @ index.4ac6a7e7.js:34
(anonymous) @ index.4ac6a7e7.js:76
(anonymous) @ index.4ac6a7e7.js:4
ie @ index.4ac6a7e7.js:4
P @ index.4ac6a7e7.js:77
(anonymous) @ index.4ac6a7e7.js:4
(anonymous) @ index.4ac6a7e7.js:4
h @ index.4ac6a7e7.js:76
```

### System Info

```shell
src="https://gradio.s3-us-west-2.amazonaws.com/3.12.0/gradio.js"
```

### Severity

blocking all usage of gradio

**Comment by krrishdholakia** [comment]
Any help here would be appreciated. 

cc: @abidlabs

**Comment by krrishdholakia** [comment]
Update, 

It seems like when I use this inside a Next.js project, it looks for a `queue/join` endpoint with the website name as the hostname

<img width="827" alt="Screen Shot 2022-12-24 at 5 58 26 PM" src="https://user-images.githubusercontent.com/17561003/209436302-74f69e8b-4f3a-46d1-9173-b221b95b6745.png">

vs. when i have it wrapped in a normal html file, it looks for it with the gradio/hf space name as the hostname 

<img width="404" alt="Screen Shot 2022-12-24 at 5 58 15 PM" src="https://user-images.githubusercontent.com/17561003/209436318-a335a275-2340-432c-9859-9b231f63490b.png">

**Comment by krrishdholakia** [comment]
For context this is happening when i'm using the gradio webcomponent in Next.js

**Comment by krrishdholakia** [comment]
This is not an error when I use the iframe. 

<img width="789" alt="Screen Shot 2022-12-24 at 6 02 33 PM" src="https://user-images.githubusercontent.com/17561003/209436392-061c55f0-d1f7-4412-b0aa-73f438501484.png">

**Comment by fabiochiusano** [comment]
Hi, I have the same problem with a Gradio app hosted on HF and embedded using Web components. The app works fine on the HF Space, works when embedded using an iframe, and doesn't work when embedded using Web components.

As @krrishdholakia said, also in my case there's a GET call to `queue/join` with my website as the hostname instead of the HF Space.

**Comment by abidlabs** [comment]
Thanks for creating the issue folks. We'll look into it! cc @aliabid94

**Comment by greendra8** [comment]
Having same issue when using <gradio-app> html tag. Happening on all gradio versions I try.

**Comment by pngwn** [comment]
Is this still present? I can't reproduce it.

https://2880-gradio-next-pngwn.vercel.app/

https://github.com/pngwn/2880-gradio-next
https://github.com/pngwn/2880-gradio-next/blob/main/src/pages/index.tsx

There is a problem in dev mode however, that I have a fix for in #3065.

**Comment by greendra8** [comment]
All my embeds are still broken when using <gradio> tag. Iframe works. Frustrating!

**Comment by pngwn** [comment]
Can i see some code please?

**Comment by fabiochiusano** [comment]
Hi! The problem is still present for me too.

The HF space I'm trying to embed: https://huggingface.co/spaces/fabiochiu/sentiment-analysis-demo/blob/main/app.py
The embed code (I tried with `3.16.2` too):
```
<script
	type="module"
	src="https://gradio.s3-us-west-2.amazonaws.com/3.14.0/gradio.js"
></script>

<gradio-app src="https://fabiochiu-sentiment-analysis-demo.hf.space"></gradio-app>
```

The embedded app raises an error after trying to call the model hosted on HF.

**Comment by pngwn** [comment]
I tried using your space and it works fine for me: https://2880-gradio-next.vercel.app/

Can you show the whole file please, or the whole project if it is public?

**Comment by pngwn** [comment]
Oh the embedding is slightly different here, let me check that.

**Comment by fabiochiusano** [comment]
Ok! Let me know if having a public page can help you, I can set it up quickly

**Comment by greendra8** [comment]
> I tried using your space and it works fine for me: https://2880-gradio-next.vercel.app/
> 
> Can you show the whole file please, or the whole project if it is public?

The demo you linked doesn't work for me and produces the error.

**Comment by pngwn** [comment]
Yeah sorry, I changed it again. I'm now able to reproduce the error.

The reason is basically that we have a number of different ways to link to a space in order to embed a gradio app on a page (too many ways really). There was some logic that incorrectly utilising the URL of the embedding page for certain requests (only when using the `src` attribute of the web component). 

I've fixed this in #3065 now and it should go out soon.

In the meantime if you use the `space` attribute instead of `src` it should work. So:

```html
<gradio-app 
  space="fabiochiu/sentiment-analysis-demo"
></gradio-app>
```

Or you can use the `host` attribute:

```html
<gradio-app 
  host="fabiochiu-sentiment-analysis-demo.hf.space"
></gradio-app>
```

We have `space`, `host` and `src` as ways to embedding a space and all are slightly different (see what i mean by too many). And I've just realised the only version of this that we link on the space ("Embed this space") is the only one that doesn't work.

So for now use one of the other methods, the fix for this will be in soon, and we will need to clean up the API for this in the future. I think we can just have one attribute that figures things our rather than 3.

Thanks for your help with this one, was sneaky!

**Comment by fabiochiusano** [comment]
That solved it for me, thank you very much!
